src/home-page/NoMatch404.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import Typography from '@material-ui/core/Typography';
+import styled from 'styled-components';
+import { Button } from '@material-ui/core';
+import { MainContainer } from '../layout/components/MainContainer';
+
+const NoMatchTypography = styled('div')`
+  display:flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction:column;
+  margin: auto;
+  min-height: 70vh;
+`;
+
+const HeaderTypography = styled(Typography)`
+  font-size:10em;
+`;
+const SubtitleTypography = styled(Typography)`
+  font-size:2em;
+  margin: ${props => props.theme.spacing(2)}px;  
+`;
+
+const StyledButton = styled(Button)`
+color: white;
+`;
+
+export const NoMatch404: React.FC<{}> = () => {
+  const history = useHistory();
+
+  return (
+    <MainContainer>
+      <NoMatchTypography>
+        <HeaderTypography variant="h4">404</HeaderTypography>
+        <SubtitleTypography variant="subtitle1">It looks like you&apos;re lost.</SubtitleTypography>
```

Comment:
Let's make use of Material Ui building blocks: instead of adding `SubtitleTypography` margin via `styled`, we can grab Mui `Box` component that offers a great API for spacing and sizing. Assuming we also move font size management to Theme, we end up with:
```
<Typography variant="h1">404</Typography>
<Box m={2}>
  <Typography variant="subtitle1">It looks like you&apos;re lost.</Typography>
</Box>
```
☝ we can remove `SubtitleTypography` and `HeaderTypography` and as a bonus we achieve a nice separation of concerns (`Box` is responsible for spacing, Typography for fonts) 🙃

Reply:
done